tools/cldr-code/src/main/java/org/unicode/cldr/util/LocaleNormalizer.java

```diff
@@ -225,4 +259,215 @@ private static LocaleSet intersectKnownWithOrgLocales(LocaleSet orgLocaleSet) {
+            } else {
+                this.solutions.put(newSolution, 1);
+            }
+        }
+
+        public void addSolutions(Map<Solution, Integer> newSolutions) {
+            for (Solution newSolution : newSolutions.keySet()) {
+                Integer newCount = newSolutions.get(newSolution);
+                if (this.solutions.containsKey(newSolution)) {
+                    this.solutions.put(newSolution, this.solutions.get(newSolution) + newCount);
+                } else {
+                    this.solutions.put(newSolution, newCount);
+                }
+            }
+        }
+
+        public void increment() {
+            ++userCount;
+        }
+    }
+
+    public static class Solution implements Comparable<Solution> {
+        public enum Type {
+            REPLACE,
+            DELETE
+        }
+
+        public final String localeId;
+        public Type type;
+        public CLDRLocale replacementLocale = null;
+
+        public Solution(String localeId, Type type, CLDRLocale repLoc) {
+            if (type != Type.REPLACE) {
+                throw new IllegalArgumentException("3-arg constructor is for REPLACE, not " + type);
+            }
+            this.localeId = localeId;
+            this.type = type;
+            this.replacementLocale = repLoc;
+        }
+
+        public Solution(String localeId, Type type) {
+            if (type != Type.DELETE) {
+                throw new IllegalArgumentException("2-arg constructor is for DELETE, not " + type);
+            }
+            this.localeId = localeId;
+            this.type = type;
+        }
+
+        @Override
+        public int compareTo(Solution o) {
+            return this.toString().compareTo(o.toString());
+        }
+
+        public String toString() {
+            if (type == null) {
+                type = Type.DELETE;
+            }
+            switch (type) {
+                case REPLACE:
+                    return type + " with " + replacementLocale.getBaseName();
+                case DELETE:
+                    return type.toString();
+                default:
+                    throw new RuntimeException("Solution type not handled: " + type);
+            }
+        }
+    }
+
+    public void checkUserLocales(String locales, LocaleSet orgLocales, ProblemMap problems) {
+        if (orgLocales == null) {
+            normalize(locales);
+        } else {
+            normalizeForSubset(locales, orgLocales);
+        }
+        // TODO: what about "en"? It's allowed by normalize(), rejected by normalizeForSubset()
+        // unless the organization has "*". But it's always read-only in Survey Tool. Maybe it
+        // should be treated similarly to default-content locales, but always deleted, or replaced
+        // by "en_XYZ"...?
+        // Reference: https://unicode-org.atlassian.net/browse/CLDR-18913
+        if (hasMessage()) {
+            rejectLocales(orgLocales, problems);
+        }
```

Comment:
does this check the SpecialLocales.java list? That already lists root and en and others